robot/impl/robot_reconfigure_weak_dependencies_test.go

```diff
@@ -285,4 +292,365 @@ func TestUpdateWeakDependents(t *testing.T) {
 	test.That(t, weak1.resources, test.ShouldHaveLength, 2)
 	test.That(t, weak1.resources, test.ShouldContainKey, base1Name)
 	test.That(t, weak1.resources, test.ShouldContainKey, base2Name)
+	test.That(t, weak1.reconfigCount, test.ShouldEqual, 3)
+}
+
+func TestWeakDependentsExplicitDependency(t *testing.T) {
+	// This test tests the behavior of weak dependents that have an explicit dependency.
+	// The robot has 3 components:
+	//   * base1 and base2: two fake base components.
+	//   * weak1: a weak component that depends on all components and also has an explicit dependency on base1.
+	logger := logging.NewTestLogger(t)
+
+	// Register a `Resource` that generates weak dependencies. Specifically instance of
+	// this resource will depend on every `component` resource. See the definition of
+	// `internal.ComponentDependencyWildcardMatcher`.
```

Comment:
I realize you copied this from elsewhere, but `internal.ComponentDependencyWildcardMatcher` is no longer a thing.

Reply:
updated